File: app/src/main/java/com/kumi/ASFramework/PageController/ASPageAnimation.java
package com.kumi.ASFramework.PageController;

import android.view.animation.Animation;
import android.view.animation.TranslateAnimation;

public class ASPageAnimation {
  private static Animation _fade_in_from_left = null;
  
  private static Animation _fade_in_from_right = null;
  
  private static Animation _fade_out_to_left = null;
  
  private static Animation _fade_out_to_right = null;
  
  public static Animation getFadeInFromLeftAnimation() {
    if (_fade_in_from_left == null) {
      _fade_in_from_left = (Animation)new TranslateAnimation(2, -1.0F, 2, 0.0F, 2, 0.0F, 2, 0.0F);
      init(_fade_in_from_left);
    } 
    return _fade_in_from_left;
  }
  
  public static Animation getFadeInFromRightAnimation() {
    if (_fade_in_from_right == null) {
      _fade_in_from_right = (Animation)new TranslateAnimation(2, 1.0F, 2, 0.0F, 2, 0.0F, 2, 0.0F);
      init(_fade_in_from_right);
    } 
    return _fade_in_from_right;
  }
  
  public static Animation getFadeOutToLeftAnimation() {
    if (_fade_out_to_left == null) {
      _fade_out_to_left = (Animation)new TranslateAnimation(2, 0.0F, 2, -1.0F, 2, 0.0F, 2, 0.0F);
      init(_fade_out_to_left);
    } 
    return _fade_out_to_left;
  }
  
  public static Animation getFadeOutTtRightAnimation() {
    if (_fade_out_to_right == null) {
      _fade_out_to_right = (Animation)new TranslateAnimation(2, 0.0F, 2, 1.0F, 2, 0.0F, 2, 0.0F);
      init(_fade_out_to_right);
    } 
    return _fade_out_to_right;
  }
  
  private static void init(Animation paramAnimation) {
    paramAnimation.setFillBefore(true);
    paramAnimation.setFillAfter(true);
    paramAnimation.setFillEnabled(true);
  }
}


/* Location:              C:\Users\kodak\Downloads\反編譯\dex-tools-v2.4\classes-dex2jar.jar!\com\kumi\ASFramework\PageController\ASPageAnimation.class
 * Java compiler version: 6 (50.0)
 * JD-Core Version:       1.1.3
 */

File: app/src/main/java/com/kota/asFramework/pageController/ASPageAnimation.kt
package com.kota.asFramework.pageController

import android.view.animation.Animation
import android.view.animation.TranslateAnimation

object ASPageAnimation {
    private var fadeInFromLeft: Animation? = null
    private var fadeInFromRight: Animation? = null
    private var fadeOutToLeft: Animation? = null
    private var fadeOutToRight: Animation? = null

    private fun init(aAnimation: Animation) {
        aAnimation.fillBefore = true
        aAnimation.fillAfter = true
        aAnimation.isFillEnabled = true
    }

    val fadeInFromLeftAnimation: Animation?
        get() {
            if (fadeInFromLeft == null) {
                fadeInFromLeft =
                    TranslateAnimation(2, -1.0f, 2, 0.0f, 2, 0.0f, 2, 0.0f)
                init(fadeInFromLeft!!)
            }
            return fadeInFromLeft
        }

    val fadeInFromRightAnimation: Animation?
        get() {
            if (fadeInFromRight == null) {
                fadeInFromRight =
                    TranslateAnimation(2, 1.0f, 2, 0.0f, 2, 0.0f, 2, 0.0f)
                init(fadeInFromRight!!)
            }
            return fadeInFromRight
        }

    val fadeOutToLeftAnimation: Animation?
        get() {
            if (fadeOutToLeft == null) {
                fadeOutToLeft =
                    TranslateAnimation(2, 0.0f, 2, -1.0f, 2, 0.0f, 2, 0.0f)
                init(fadeOutToLeft!!)
            }
            return fadeOutToLeft
        }

    val fadeOutTtRightAnimation: Animation?
        get() {
            if (fadeOutToRight == null) {
                fadeOutToRight =
                    TranslateAnimation(2, 0.0f, 2, 1.0f, 2, 0.0f, 2, 0.0f)
                init(fadeOutToRight!!)
            }
            return fadeOutToRight
        }
}
